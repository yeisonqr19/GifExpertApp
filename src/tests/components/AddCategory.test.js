import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";
describe("Pruebas en <AddCategory />", () => {
  const setCategories = jest.fn();

  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories} />);

    jest.clearAllMocks();
  });
  const value = "Hola Mundo";

  test("El componente debe renderizarse perfectamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de Cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });
  });

  //   test("El parrafo debe coincidir con el argumento que yo le estoy enviando", () => {
  //     expect(wrapper.find("p").text().trim()).toBe(value);
  //   });

  test("No debe de Postear la informacion con submit ", () => {
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar el setCategories y limpiar la caja de texto", () => {

    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    const input = wrapper.find("input");

    input.simulate("change", { target: { value } });
    expect(input.props().value).toBe("");

    wrapper.find("form").simulate("submit", { preventDefault: () => {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    // expect(setCategories).toHaveBeenCalledWith( expect.any(function) );

  });
});
