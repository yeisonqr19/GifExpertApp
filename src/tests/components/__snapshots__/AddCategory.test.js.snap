// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> El componente debe renderizarse perfectamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
